tests/test_path_resolution.py

```diff
@@ -184,3 +184,28 @@ def handler_fn(a: int = 0, b: int = 0, c: int = 0, d: int = 0) -> None:
 
     with pytest.raises(ImproperlyConfiguredException):
         create_test_client(handler_fn)
+
+
+@pytest.mark.parametrize(
+    "handler_path, request_path, expected_status_code, expected_param",
+    [
+        ("/name:str/{name:str}", "/name:str/test", HTTP_200_OK, "test"),
+        ("/user/*/{name:str}", "/user/foo/bar", HTTP_404_NOT_FOUND, None),
+    ],
+)
+def test_special_chars(
+    handler_path: str, request_path: str, expected_status_code: int, expected_param: Optional[str]
+) -> None:
+    @get(path=handler_path, media_type=MediaType.TEXT)
+    def handler_fn(name: str) -> str:
+        return name
+
+    with create_test_client(handler_fn) as client:
+        response = client.get(request_path)
+        assert response.status_code == expected_status_code
+
+        if response.ok:
+            if expected_param is not None:
+                assert response.text == expected_param
+            else:
+                assert response.content == b""
```

Comment:
what are you testing here and how does this relate to the test case🙃 ?

Reply:
A check to make sure the parameter was correctly parsed from the request URL, which is returned by the test route handler. Previously, "/*/{name:str}" was parsed as two path parameters, meaning a request for "/foo/bar" would return "foo"